packages/design-system/src/components/OcResourceIcon/OcResourceIcon.vue

```diff
@@ -4,24 +4,27 @@
     :name="iconName"
     :color="iconColor"
     :size="size"
-    :class="[
-      'oc-resource-icon',
-      { 'oc-resource-icon-file': !isFolder, 'oc-resource-icon-folder': isFolder }
-    ]"
+    :class="['oc-resource-icon', iconTypeClass]"
   />
 </template>
 
-<script>
+<script lang="ts">
+import { defineComponent, PropType } from 'vue'
+import { Resource } from 'web-client'
+
 import OcIcon from '../OcIcon/OcIcon.vue'
-import iconNameMap from '../../helpers/resourceIconExtensionMapping'
-import iconColorMap from '../../helpers/resourceIconColorExtensionMapping'
+import { AVAILABLE_SIZES } from '../../helpers/constants'
```

Comment:
```suggestion
import { AVAILABLE_SIZES } from '../../helpers'
```

(at least that's what my IDE is complaining about ;-) )

Reply:
It's possible but can eventually lead to circular import (stumbled upon this last week), will fix for now but keep in mind pls :)